src/test/java/org/cactoos/io/HeadInputTest.java

```diff
@@ -0,0 +1,384 @@
+    }
+
+    @Test
+    public void readsEncodedStringCharSequence() throws Exception {
+        final int length = 3;
+        final CharSequence charsequence = "CactoosEncoded";
+        MatcherAssert.assertThat(
+            "Can't read encoded string charsequence of bytes",
+            new String(
+                new BytesOf(
+                    new HeadInput(
+                        charsequence,
+                        "UTF-8",
+                        length
+                    )
+                ).asBytes()
+            ),
+            Matchers.equalTo("Cac")
+        );
+    }
+
+    @Test
+    public void readsText() {
+        final int length = 5;
+        MatcherAssert.assertThat(
+            "Can't read string from text",
+            new TextOf(
+                new HeadInput(
+                    new TextOf("CactoosOOP"), length
+                )
+            ),
+            new TextHasString(Matchers.containsString("Cacto"))
+        );
+    }
+
+    @Test
+    public void readsEncodedText() {
+        final int length = 2;
+        MatcherAssert.assertThat(
+            "Can't read string with enum charset from text",
+            new TextOf(
+                new HeadInput(
+                    new TextOf("CactoosLibrary"), StandardCharsets.UTF_8, length
+                )
+            ),
+            new TextHasString(Matchers.containsString("Ca"))
+        );
+    }
+
+    @Test
+    public void readsEncodedStringText() {
+        final int length = 8;
+        final String charset = "UTF-8";
+        MatcherAssert.assertThat(
+            "Can't read string with literal charset from text",
+            new TextOf(
+                new HeadInput(
+                    new TextOf("CactoosText"), charset, length
+                )
+            ),
+            new TextHasString(Matchers.containsString("CactoosT"))
+        );
+    }
+
+    @Test
+    public void readsAnArrayOfBytes() throws Exception {
+        final int length = 1;
+        final byte[] bytes = new byte[]{(byte) 0xCA, (byte) 0xFE};
+        MatcherAssert.assertThat(
+            "Can't read array of bytes",
+            new InputAsBytes(
+                new SyncInput(new HeadInput(bytes, length))
+            ).asBytes(),
+            Matchers.equalTo(new byte[]{bytes[0]})
+        );
+    }
+
+    @Test
+    public void readsSimpleFileContent() throws IOException {
+        final int length = 7;
+        final Path temp = Files.createTempFile("cactoos-1", "txt-1");
+        final String content = "Hello, товарищ!";
+        Files.write(temp, content.getBytes(StandardCharsets.UTF_8));
+        MatcherAssert.assertThat(
+            "Can't read file content",
+            new HeadInput(temp, length),
+            new InputHasContent(content.substring(0, length))
+        );
+    }
+
+    @Test
+    public void readsFromInputStream() throws IOException {
+        final int length = 3;
+        final Path temp = Files.createTempFile("cactoos-2", "txt-2");
```

Comment:
@memoyil  Please don't use `Files.createTempFile`, we are trying to merge PR #779 wich replace all of this with `TemporaryFolder`.